Authenticator/Source/TokenManager.swift

```diff
@@ -2,93 +2,71 @@
-        }
-        set {
-            let defaults = NSUserDefaults.standardUserDefaults()
-            defaults.setObject(newValue, forKey: kOTPKeychainEntriesArray)
-            defaults.synchronize()
-        }
-    }
-
-    private func fetchTokensFromKeychain() {
-        keychainItems = TokenManager.keychainItems(Token.KeychainItem.allKeychainItems(),
-            sortedByPersistentRefs: keychainItemRefs)
-
-        if keychainItems.count > keychainItemRefs.count {
-            // If lost tokens were found and appended, save the full list of tokens
-            saveTokenOrder()
-        }
-    }
-
-    private class func keychainItems(keychainItems: [Token.KeychainItem],
-        sortedByPersistentRefs persistentRefs: [NSData]) -> [Token.KeychainItem]
-    {
-        var sorted: [Token.KeychainItem] = []
-        var remaining = keychainItems
-        // Iterate through the keychain item refs, building an array of the corresponding tokens
-        for persistentRef in persistentRefs {
-            let indexOfTokenWithSameKeychainItemRef = remaining.indexOf {
-                return ($0.persistentRef == persistentRef)
-            }
-
-            if let index = indexOfTokenWithSameKeychainItemRef {
-                let matchingItem = remaining[index]
-                remaining.removeAtIndex(index)
-                sorted.append(matchingItem)
+        do {
+            let persistentTokenSet = try keychain.allPersistentTokens()
+            let sortedIdentifiers = TokenManager.persistentIdentifiers()
+
+            persistentTokens = persistentTokenSet.sort({ (A, B) in
+                let indexOfA = sortedIdentifiers.indexOf(A.identifier)
+                let indexOfB = sortedIdentifiers.indexOf(B.identifier)
+
+                switch (indexOfA, indexOfB) {
+                case (.Some(let iA), .Some(let iB)) where iA < iB:
+                    return true
+                default:
+                    return false
+                }
+            })
+
+            if persistentTokens.count > sortedIdentifiers.count {
+                // If lost tokens were found and appended, save the full list of tokens
+                saveTokenOrder()
             }
+        } catch {
+            persistentTokens = []
+            // TODO: Handle the token loading error
         }
-        // Append the remaining tokens which didn't match any keychain item refs
-        return sorted + remaining
     }
 
     // MARK: -
 
     var numberOfTokens: Int {
-        return keychainItems.count
+        return persistentTokens.count
     }
 
     /// Returns a sorted, uniqued array of the periods of timer-based tokens
     var timeBasedTokenPeriods: [NSTimeInterval] {
-        let periods = keychainItems.reduce(Set<NSTimeInterval>()) { (var periods, keychainItem) in
-            if case .Timer(let period) = keychainItem.token.generator.factor {
+        let periods = persistentTokens.reduce(Set<NSTimeInterval>()) { (var periods, persistentToken) in
```

Comment:
Length Violation: Line should be 100 characters or less: currently 104 characters